### ISSUE TITLE
npm install error on Intel Edison Yocto release 2.1

### ISSUE THREAD

**Issue by Vayvala** [intel-iot-devkit/mraa#272]
Intel Edison freshly updated to Yocto release 2.1

```
root@va:~/js# uname -a
Linux va 3.10.17-poky-edison+ #1 SMP PREEMPT Fri Jun 19 12:06:40 CEST 2015 i686 GNU/Linux
root@va:~/js# npm install --verbose mraa
..........
  cc '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DSWIG' '-DSWIGJAVASCRIPT' '-DBUILDING_NODE_EXTENSION=1' '-DSWIG_V8_VERSION=0x031459' '-DV8_VERSION=0x031459' '-DBUILDING_NODE_EXTENSION' -I/home/root/0.10.38/src -I/home/root/0.10.38/deps/uv/include -I/home/root/0.10.38/deps/v8/include -I../include -I../api/mraa -I../api  -Wall -Wextra -Wno-unused-parameter -pthread -m32 -O2 -fno-strict-aliasing -fno-tree-vrp -fno-omit-frame-pointer  -MMD -MF ./Release/.deps/Release/obj.target/mraa/src/mraa.o.d.raw  -c -o Release/obj.target/mraa/src/mraa.o ../src/mraa.c
../src/mraa.c: In function 'mraa_init':
../src/mraa.c:110:2: error: #error mraa_ARCH NOTHING
 #error mraa_ARCH NOTHING
  ^
../src/mraa.c: In function 'mraa_set_priority':
../src/mraa.c:140:18: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (priority > sched_get_priority_max(SCHED_RR)) {
                  ^
../src/mraa.c: In function 'mraa_setup_mux_mapped':
../src/mraa.c:154:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (mi = 0; mi < meta.mux_total; mi++) {
                     ^
../src/mraa.c: In function 'mraa_pin_mode_test':
../src/mraa.c:233:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (pin > (plat->phy_pin_count - 1) || pin < 0)
             ^
../src/mraa.c: In function 'mraa_get_pin_name':
../src/mraa.c:353:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (pin > (plat->phy_pin_count - 1) || pin < 0)
             ^
../src/mraa.c: In function 'mraa_find_i2c_bus':
../src/mraa.c:520:9: warning: statement with no effect [-Wunused-value]
         for (i; i < num_i2c_devices; i++) {
         ^
mraa.target.mk:110: recipe for target 'Release/obj.target/mraa/src/mraa.o' failed
make: *** [Release/obj.target/mraa/src/mraa.o] Error 1
make: Leaving directory '/home/root/js/node_modules/mraa/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Linux 3.10.17-poky-edison+
gyp ERR! command "node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/root/js/node_modules/mraa
gyp ERR! node -v v0.10.38
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok

npm verb unsafe-perm in lifecycle false
npm info mraa@0.7.4 Failed to exec install script
npm ERR! mraa@0.7.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mraa@0.7.4 install script.
npm ERR! This is most likely a problem with the mraa package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls mraa
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.10.17-poky-edison+
npm ERR! command "node" "/usr/bin/npm" "install" "--verbose" "mraa"
npm ERR! cwd /home/root/js
npm ERR! node -v v0.10.38
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm verb exit [ 1, true ]
npm verb node_modules/mraa unbuild
npm info preuninstall mraa@0.7.4
npm info uninstall mraa@0.7.4
npm verb true,/home/root/js/node_modules,/home/root/js/node_modules unbuild mraa@0.7.4
npm info postuninstall mraa@0.7.4
npm ERR! not ok code 0
```

**Comment by arfoll** [comment]
NPM module is broken currently on node.js <0.12.0. I'll release a new version asap to fix this.

**Comment by arfoll** [comment]
This'll need a new release which I'll do later today to make sure it's in npm.

**Comment by arfoll** [comment]
Should be fixed with 0.7.5 that's uploaded. Tested on node 0.10.x both on 64 and 32bit.
